chapters/VK_EXT_external_memory_metal/device_memory.adoc

```diff
@@ -0,0 +1,153 @@
+
+=== Metal External Memory
+
+[open,refpage='VkImportMemoryMetalHandleInfoEXT',desc='Import Metal memory created on the same physical device',type='structs']
+--
+To import memory from a Metal handle, add a
+slink:VkImportMemoryMetalHandleInfoEXT structure to the pname:pNext chain of
+the slink:VkMemoryAllocateInfo structure.
+
+The sname:VkImportMemoryMetalHandleInfoEXT structure is defined as:
+
+include::{generated}/api/structs/VkImportMemoryMetalHandleInfoEXT.adoc[]
+
+  * pname:sType is a elink:VkStructureType value identifying this structure.
+  * pname:pNext is `NULL` or a pointer to a structure extending this
+    structure.
+  * pname:handleType is a elink:VkExternalMemoryHandleTypeFlagBits value
+    specifying the type of pname:handle or pname:name.
+  * pname:handle is `NULL` or the external handle to import.
+
+Importing memory object payloads from Metal handles shares the
+ownership of the handle to the Vulkan implementation.
+
+Applications can: import the same payload into multiple instances of Vulkan,
+into the same instance from which it was exported, and multiple times into a
+given Vulkan instance.
+In all cases, each import operation must: create a distinct
+sname:VkDeviceMemory object.
+
+.Valid Usage
+****
+  * If pname:handleType is not `0`, it must: be supported for import, as
+    reported by slink:VkExternalImageFormatProperties or
+    slink:VkExternalBufferProperties
+  * The memory from which pname:handle was exported must: have been created
+    on the same underlying physical device as pname:device
+  * If pname:handleType is not `0`, it must: be
+    ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLBUFFER_BIT_EXT,
+    ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLTEXTURE_BIT_EXT or
+    ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLHEAP_BIT_EXT
+  * If pname:handleType is not `0` , pname:handle must: be a valid non-NULL
+    handle of the type specified by pname:handleType
+  * pname:handle must: obey any requirements listed for pname:handleType in
+    <<external-memory-handle-types-compatibility,external memory handle
+    types compatibility>>
+****
+
+include::{generated}/validity/structs/VkImportMemoryMetalHandleInfoEXT.adoc[]
+--
+
+[open,refpage='vkGetMemoryMetalHandleEXT',desc='Get a Metal handle for a memory object',type='protos']
+--
+To export a Metal handle representing the payload of a Vulkan device
+memory object, call:
+
+include::{generated}/api/protos/vkGetMemoryMetalHandleEXT.adoc[]
+
+  * pname:device is the logical device that created the device memory being
+    exported.
+  * pname:pGetMetalHandleInfo is a pointer to a
+    slink:VkMemoryGetMetalHandleInfoEXT structure containing parameters of
+    the export operation.
+  * pname:pHandle will return the Metal handle representing the payload of
+    the device memory object.
+
+The lifespan of the returned handle object is the same as the associated `VkDeviceMemory` object.
+The app must: retain the handle object if it intends to use it beyond that lifespan.
```

Comment:
I feel that the lifetime language is incorrect, unless I misunderstand the meaning of "retain". The text says that the lifespan of the Metal handle is the same as the VkDeviceMemory; therefore, destruction of the VkDeviceMemory also invalidates the Metal handle. However, the text's sentence suggests that the lifetime can be extended by "retaining" the handle; this sentence contradicts the first sentence.

Reply:
Metal objects are reference counted and only destroyed when the count reaches 0. This part of the spec is trying to state that unless the user calls `retain` (which increases the count) on the returned Metal object, the lifespan will be that of the `VkDeviceMemory`. That is why `beyond that lifespan` is used.

I will try to think of a better way to word this part since I see it can lead to confusion. If you have any suggestion on how to word it, do let me know!

Reply:
Let me know if it's clearer now!